src/common/verify/isValidAccessGrant.ts

```diff
@@ -48,7 +51,20 @@ async function isValidAccessGrant(
   } = {},
 ): Promise<{ checks: string[]; warnings: string[]; errors: string[] }> {
   const fetcher = await getSessionFetch(options);
-  const vcObject = await getBaseAccess(vc, options);
+  let validVC = null;
+  if (
+    typeof vc !== "string" &&
+    !isUrl(vc.toString()) &&
+    !(vc instanceof URL) &&
+    !isDatasetCore(vc)
+  ) {
+    validVC = await verifiableCredentialToDataset(vc, {
+      includeVcProperties: true,
+      requireId: false,
+    });
+  }
```

Comment:
```suggestion
  if (
    typeof vc === "object" &&
    !isDatasetCore(vc) &&
    typeof (vc as { id: string }).id === "string"    
  ) {
    validVC = await verifiableCredentialToDataset(vc, {
      includeVcProperties: true,
      requireId: true,
    });
  }
```
I think I would prefer the logic here shifted to check whether the object that got passed is indeed a plain JSON representation of a Verifiable Credential, which would have a `.id` property. It is more restrictive than going by elimination, and assuming that a non-string non-URL non-Datasetcore object would be parseable as JSON-LD.

Reply:
Following the prior review, this change in the validations would be in `toVcDataset`?

Reply:
Absolutely

Reply:
So, just to sum up, `vcToDataset` will either fetch the vc if it is a valid vc, that is:

`    if (typeof vc === "string" || vc instanceof URL) {
        return await getVerifiableCredential(vc.toString(), {
            returnLegacyJsonld: false,
            skipValidation: true,
            fetch: options.fetch,
        });
    }`

or, if it is not, convert it to a valid vc:

`if (
        typeof vc === "object" && 
        !isDatasetCore(vc) && 
        typeof (vc as { id: string }).id === "string"
    ) {
        return await verifiableCredentialToDataset(vc, {
            includeVcProperties: true,
            requireId: true,
        });
    }`